### PR TITLE
Added support of pull down to refresh feature for Android platform.

### PR DESCRIPTION
**Pull request by muskan273**

### CODE REVIEW

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 1]
```diff
@@ -730,23 +775,24 @@ Map<String, Integer> getCommandsMap() {
   }
 
   @Override
-  public void receiveCommand(@NonNull WebView root, String commandId, @Nullable ReadableArray args) {
+  public void receiveCommand(@NonNull LinearLayout root, String commandId, @Nullable ReadableArray args) {
+    Log.d(TAG, "receiveCommand: " + root.getId());
```
Comment (Kasea) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:779]: Do we want to leave this in?

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 2]
```diff
@@ -1238,6 +1296,8 @@ protected static class RNCWebChromeClient extends WebChromeClient implements Lif
     protected List<String> pendingPermissions = new ArrayList<>();
 
     protected RNCWebView.ProgressChangedFilter progressChangedFilter = null;
+    // True if protected media should be allowed, false otherwise
+    protected boolean mAllowsProtectedMedia = false;
```
Comment (Kasea) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:1300]: duplicate
```suggestion
```

**Review by Kasea — CHANGES_REQUESTED** [review]
First, I love how "fast" react-native libraries/developers are when it comes to "bugs" that have existed in their libraries for years that are a huge deal breaker 😄.

Getting to the PR though; There are a couple of problems which are a "deal breaker" (imo).

When dealing with a public project or any project for that matter, ALWAYS conform to the code bases code style/practices.
This includes how they leave comments, indentation, ordering of methods (although it might be random, see next point), method/property naming and everything else inbetween.
You've mostly done this, although there are places where you've changed indentation (to the correct one technically, but see next point) and added spaces.

When dealing with a public project, always keep the PRs as small as possible. I can see you've "fixed" a lot of formatting "errors", but this just adds to more code to review. Keep it as simple as possible. If this was at a company I would be happy to see some code cleanup, but these developers are not paid and want to do the least possible 😓 . You've also moved around a couple of methods and in some cases duplicated them. Don't do unnecessary work!

**Comment by Kasea** [comment]
NOTE: if you have a modal with overflow, scroll down on said modal, release and then try to scroll up, it'll refresh instead of scrolling up.

**Comment by snaptravel-eng** [comment]
@Kasea @muskan273 What do we need to do to get this merged in? Is the only issue the code style/formatting or is the modal with overflow a dealbreaker?

**Comment by stanimirsellercloud** [comment]
@snaptravel-eng I think this also fixes pull to refresh on Android.

**Comment by snaptravel-eng** [comment]
Thanks. Yeah I understand the intention. How do we get this merged instead of stuck as a PR?

**Comment by stanimirsellercloud** [comment]
cc @Titozzz @Saadnajmi

**Comment by rafaelmonroy** [comment]
does anyone know when this is getting merged? 👀 @muskan273 @Kasea @Titozzz @Saadnajmi

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 1 above)
Comment (muskan273) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Only need to remove the log

**Comment by muskan273** [comment]
@Kasea Code modified & pushed as per the above suggestions, Can you please review it and let me know if I need to make any changes

**Comment by snaptravel-eng** [comment]
Thanks for updating @muskan273 !

How do we get this merged instead of stuck as a PR? @Kasea @jamonholmgren @Titozzz

**Comment by muskan273** [comment]
> NOTE: if you have a modal with overflow, scroll down on said modal, release and then try to scroll up, it'll refresh instead of scrolling up.
> 
> This is working as intended on iOS

**Comment by muskan273** [comment]
> NOTE: if you have a modal with overflow, scroll down on said modal, release and then try to scroll up, it'll refresh instead of scrolling up.
> 
> This is working as intended on iOS

I faced the same issue while implementing that feature, you will find the solution mentioned in the below blog:
 https://dev.to/muskanparashar/react-native-webview-make-android-pulltorefresh-work-without-any-glitches-467

**Comment by peterlazar1993** [comment]
@muskan273 Congrats on your first PR!
I'd love to test this, could you please rebase master.

**Comment by Muneeb926595** [comment]
is there anyone who can please rebase master please i need this PR urgently.

**Comment by snaptravel-eng** [comment]
Is there any blocker for updating and merging this? The open bounty still applies: https://github.com/react-native-webview/react-native-webview/issues/2678
Thanks

**Comment by StephenCavender** [comment]
this pr doesn't work for loading web pages that have nested scroll views. The page i'm trying to load has a scrollable content block between a sticky header and footer. I can't scroll up within the scrollable content, it just triggers the refresh.

Repro is to replace the example for [Scrolling](https://github.com/muskan273/react-native-webview/blob/feature/pull_to_refresh_android/example/examples/Scrolling.tsx) with this: 
```
import React, { Component } from 'react';
import { Button, Text, View } from 'react-native';

import WebView from 'react-native-webview';

const HTML = `
<!DOCTYPE html>\n
<html>
  <head>
    <title>Hello World</title>
    <meta http-equiv="content-type" content="text/html; charset=utf-8">
    <meta name="viewport" content="width=320, user-scalable=no">
    <style type="text/css">
    #flexbox_container {
      display: flex;
      flex-direction: column;
      box-shadow: 0 4px 5px -2px #999;
      height: 90%;
    }
    
    .flex_item {
      background: #eee;
      margin: 4px;
      padding: 10px;
    }
    
    .middle {
      flex-grow: 1;
      overflow: auto;
    }
    
    html,
    body {
      height: 100%;
    }
    </style>
  </head>
  <body>
  <div id="flexbox_container">
  <div class="flex_item">
    Item</div>
  <div class="flex_item middle">
    <p>1</p>
    <p>2</p>
    <p>3</p>
    <p>4</p>
    <p>5</p>
    <p>6</p>
    <p>7</p>
    <p>8</p>
    <p>9</p>
    <p>10</p>
    <p>11</p>
    <p>12</p>
    <p>13</p>
  </div>
  <div class="flex_item">Item</div>
</div>
  </body>
</html>
`;

type Props = {};
type State = {
  scrollEnabled: boolean;
  lastScrollEvent: string;
};

export default class Scrolling extends Component<Props, State> {
  state = {
    scrollEnabled: true,
    lastScrollEvent: '',
  };

  render() {
    return (
      <View>
        <View style={{ height: 240 }}>
          <WebView
            source={{ html: HTML }}
            automaticallyAdjustContentInsets={false}
            onScroll={this._onScroll}
            scrollEnabled={this.state.scrollEnabled}
          />
        </View>
        <Button
          title={
            this.state.scrollEnabled ? 'Scroll enabled' : 'Scroll disabled'
          }
          onPress={() =>
            this.setState({ scrollEnabled: !this.state.scrollEnabled })
          }
        />
        <Text>Last scroll event:</Text>
        <Text>{this.state.lastScrollEvent}</Text>
      </View>
    );
  }

  _onScroll = (event) => {
    this.setState({ lastScrollEvent: JSON.stringify(event.nativeEvent) });
  };
}


```
Recording of behavior:
https://user-images.githubusercontent.com/6619472/235944464-2d826e6e-4ef2-40ff-853b-a682b6f54169.mov

**Comment by muskan273** [comment]
Hi, I wanted to bring your attention back to a closed pull request that I believe requires further consideration. I have identified a potential solution for the issue discussed in this comment: https://github.com/react-native-webview/react-native-webview/pull/2787#issuecomment-1533117955.
 You can review the proposed solution in the attached file provided below:

[WebviewRefresh.js.zip](https://github.com/react-native-webview/react-native-webview/files/12405721/WebviewRefresh.js.zip)

Looking forward to your feedback and the possibility of reopening and resolving this pull request.
@Kasea @StephenCavender @snaptravel-eng @stanimirsellercloud @rafaelmonroy @peterlazar1993

**Pull request closed without merging**